db/migrate/20200331150436_rename_foreman_features.rb

```diff
@@ -0,0 +1,77 @@
+class RenameForemanFeatures < ActiveRecord::Migration[5.1]
+  class MiqProductFeature < ActiveRecord::Base; end
+  class MiqRolesFeature < ActiveRecord::Base; end
+  class User < ActiveRecord::Base
+    serialize :settings, Hash
+  end
+
+  FEATURE_MAPPING = {
+    'providers_accord'                   => 'ems_configuration',
+    'provider_foreman_view'              => 'ems_configuration_view',
+    'provider_foreman_control'           => 'ems_configuration_control',
+    'configuration_manager_provider_tag' => 'ems_configuration_tag',
+    'provider_admin'                     => 'ems_configuration_admin',
+    'provider_foreman_delete_provider'   => 'ems_configuration_delete_provider',
+    'provider_foreman_edit_provider'     => 'ems_configuration_edit_provider',
+    'provider_foreman_add_provider'      => 'ems_configuration_add_provider',
+    'configured_systems_filter_accord'   => 'configured_system',
+  }.freeze
+
+  ROLES_FEATURE_MAPPING = {
+    'provider_foreman_explorer'             => %w[ems_configuration configured_system configuration_profile],
+    'configured_systems_filter_accord'      => %w[configured_system],
+    'configured_systems_filter_accord_view' => %w[configured_system_view],
+    'configuredsystem_control'              => %w[configured_system_control],
+  }.freeze
+
+  def up
+    return if MiqProductFeature.none?
+
+    # Direct renaming of features
+    FEATURE_MAPPING.each do |from, to|
```

Comment:
For data migrations, we also tend to use `say_with_time` to give feedback to the user about how long it takes.

Not sure of how granular each block should be,  It could be one giant block around the whole thing, but it might be better to have 3 sections  since you've broken up the code that way as well.